pkg/controller/metrics/metrics.go

```diff
@@ -18,14 +18,90 @@ import (
 	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
 	hiveintv1alpha1 "github.com/openshift/hive/pkg/apis/hiveinternal/v1alpha1"
 	"github.com/openshift/hive/pkg/constants"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
 	"github.com/openshift/hive/pkg/imageset"
 )
 
 const (
 	ControllerName = hivev1.MetricsControllerName
 )
 
+// metrics collected through custom prometheus collector
+type customCollector struct {
```

Comment:
lets rename to provisioningUnderwayCollector and godoc appropriately as well.

Reply:
also we should probably break this out into a separate provisionunderwaycollector.go module in this package.

Reply:
Changed.
FWIW my intention was to keep it abstract - so we can easily adapt it to add more metrics this way.